src/plugins/visualizations/public/vis_types/types_service.ts

```diff
@@ -17,71 +17,48 @@
  * under the License.
  */
 
-import { IconType } from '@elastic/eui';
 import { visTypeAliasRegistry, VisTypeAlias } from './vis_type_alias_registry';
 import { BaseVisType, BaseVisTypeOptions } from './base_vis_type';
 import { ReactVisType, ReactVisTypeOptions } from './react_vis_type';
-import { TriggerContextMapping } from '../../../ui_actions/public';
-
-export interface VisType {
-  name: string;
-  title: string;
-  description?: string;
-  getSupportedTriggers?: () => Array<keyof TriggerContextMapping>;
-  visualization: any;
-  isAccessible?: boolean;
-  requestHandler: string | unknown;
-  responseHandler: string | unknown;
-  icon?: IconType;
-  image?: string;
-  stage: 'experimental' | 'beta' | 'production';
-  requiresSearch: boolean;
-  hidden: boolean;
-
-  // Since we haven't typed everything here yet, we basically "any" the rest
-  // of that interface. This should be removed as soon as this type definition
-  // has been completed. But that way we at least have typing for a couple of
-  // properties on that type.
-  [key: string]: any;
-}
+import { VisType } from './types';
 
 /**
  * Vis Types Service
  *
  * @internal
  */
 export class TypesService {
-  private types: Record<string, VisType> = {};
+  private types: Record<string, VisType<any>> = {};
   private unregisteredHiddenTypes: string[] = [];
 
-  public setup() {
-    const registerVisualization = (registerFn: () => VisType) => {
-      const visDefinition = registerFn();
-      if (this.unregisteredHiddenTypes.includes(visDefinition.name)) {
-        visDefinition.hidden = true;
-      }
+  private registerVisualization<TVisParam>(visDefinition: VisType<TVisParam>) {
```

Comment:
:information_source: Just pulling this into an actual class method, instead of defining it as a closure, since it did not needed any access to anything that closure would have had.